src/example.zig

```diff
@@ -71,60 +87,38 @@ fn main_generic(allocator: std.mem.Allocator) Editor.Error!void {
         defer allocator.free(line);
 
         try editor.addToHistory(line);
-        std.log.info("line ({} bytes): {s}\n", .{ line.len, line });
+        std.log.info("line ({d} bytes): {s}\n", .{ line.len, line });
 
         if (std.mem.eql(u8, line, "quit")) {
             break;
         }
     }
 }
 
-const WriterContext = struct {
-    console_out: *std.os.uefi.protocol.SimpleTextOutput,
-    attribute: usize,
-};
-
-fn writeFn(context: *const anyopaque, bytes: []const u8) anyerror!usize {
-    const writer_context: *const WriterContext = @alignCast(@ptrCast(context));
-    const console_out = writer_context.console_out;
-    _ = console_out.setAttribute(writer_context.attribute);
-    for (bytes) |c| {
-        if (c == '\n') {
-            _ = console_out.outputString(@ptrCast(&[2]u16{ '\r', 0 }));
-        }
-        _ = console_out.outputString(@ptrCast(&[2]u16{ c, 0 }));
-    }
-    return bytes.len;
-}
-
-fn main_uefi(allocator: std.mem.Allocator) void {
+fn main_uefi(allocator: std.mem.Allocator) !void {
     var editor = Editor.init(allocator, .{});
     defer editor.deinit();
 
-    // kiesel: src/branch/main/src/uefi.zig:69
+    // kiesel: src/branch/main/src/uefi.zig
     const console_out = std.os.uefi.system_table.con_out.?;
-    _ = console_out.reset(true);
-    _ = console_out.clearScreen();
+    try console_out.reset(true);
+    try console_out.clearScreen();
+    try console_out.enableCursor(true);
 
-    const stdout: std.io.AnyWriter = .{
-        .context = &WriterContext{
-            .console_out = console_out,
-            .attribute = 0x7, // white
-        },
-        .writeFn = writeFn,
-    };
+    var stdout_buffer: [1024]u8 = undefined;
```

Comment:
I didn't spend too much thought on these buffer sizes, 1kB seems to be a common default. (the old removed buffered writer used 4kB by default)